cmd/kube-controller-manager/app/options/csrsigningcontroller.go

```diff
@@ -46,6 +46,7 @@ func (o *CSRSigningControllerOptions) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&o.ClusterSigningCertFile, "cluster-signing-cert-file", o.ClusterSigningCertFile, "Filename containing a PEM-encoded X509 CA certificate used to issue cluster-scoped certificates")
 	fs.StringVar(&o.ClusterSigningKeyFile, "cluster-signing-key-file", o.ClusterSigningKeyFile, "Filename containing a PEM-encoded RSA or ECDSA private key used to sign cluster-scoped certificates")
 	fs.DurationVar(&o.ClusterSigningDuration.Duration, "experimental-cluster-signing-duration", o.ClusterSigningDuration.Duration, "The length of duration signed certificates will be given.")
+	fs.BoolVar(&o.ClusterSigningReturnSigner, "cluster-signing-return-signer", o.ClusterSigningReturnSigner, "If true, the controller will also return the signing CA PEM, appended to the certificate body.")
```

Comment:
If a user has a PKI hierarchy that goes `root -> intermediate 1 -> intermediate 2 -> leaf` is it expected that they include `intermediate 2 -> intermediate 1` in their `cluster-signing-cert-file` so that the issued certs go `leaf -> intermediate 2 -> intermediate 1`?

Otherwise, we need another explicit flag (i.e. `cluster-signing-cert-chain`) or some form of `depth` parameter (which seems more opaque than an explicit flag IMO).

I know some of this was discussed by yourself on Slack, but raising it here for posterity 😄

Reply:
I agree. I think explicitly providing the intermediate chain to append is preferred.

Reply:
> explicitly providing the intermediate chain to append is preferred

Agree as well.

Reply:
>  I think explicitly providing the intermediate chain to append is preferred over a boolean "append the signing cert" option.
This is sensible, will do

Reply:
Small problem with this @enj @liggitt @munnerz, if we add a separate chain file and load it separately, it can be loaded at a different time to the cert file, and we could end up out of sync. I think it may be preferable to use a bool, include signers, and down the line we can allow the user to provide a chain as `ClusterSigningCertFile`?